public/apps/configuration/app-router.tsx

```diff
@@ -143,140 +144,154 @@ function decodeParams(params: { [k: string]: string }): any {
   }, {});
 }
 
+export interface DataSourceContextType {
+  dataSource: DataSourceOption;
+  setDataSource: React.Dispatch<React.SetStateAction<DataSourceOption>>;
+}
+
+export const DataSourceContext = createContext<DataSourceContextType | null>(null);
+
 export function AppRouter(props: AppDependencies) {
   const setGlobalBreadcrumbs = flow(getBreadcrumbs, props.coreStart.chrome.setBreadcrumbs);
+  const [dataSource, setDataSource] = useState<DataSourceOption>({
+    id: '',
+    label: 'Local cluster',
```

Comment:
Is there a convention to use to refer to the `Default OpenSearch`? Local cluster may be applicable for testing, but would should this value really be?

Reply:
Yes this is what is coming from core - ref: https://github.com/opensearch-project/OpenSearch-Dashboards/blob/902e216765f0ed232861a8a883e0799d30af21d6/src/plugins/data_source_management/public/components/data_source_selector/data_source_selector.tsx#L13, but I think you are onto something here in that we shouldn't replicate existing functionality, so imported it. Thanks!

Reply:
@cwperks following up here - It is actually not possible to import it from core - see failed run here: https://github.com/opensearch-project/security-dashboards-plugin/actions/runs/8395773325/job/22995781261. Issue is that core is not exporting this for reference. Reverted it to the original code, but that is matching the "Local cluster" coming from core.